src/data.rs

```diff
@@ -40,35 +41,47 @@ pub async fn fetch_cells_from_ipfs(
 	positions: &Vec<DataCell>,
 ) -> Result<(Vec<DataCell>, Vec<DataCell>)> {
 	// TODO: Should we fetch peers before loop or for each cell?
-	let peers = ipfs.peers();
+	let peers = &ipfs.peers();
 	log::debug!("Number of known IPFS peers: {}", peers.len());
 
 	if peers.is_empty() {
 		log::info!("No known IPFS peers");
 		return Ok((vec![], positions.to_vec()));
 	}
 
-	let mut fetched = vec![];
-	let mut unfetched = vec![];
-	// TODO: Paralelize fetching
-	for position in positions {
-		match fetch_cell_from_ipfs(ipfs, peers.clone(), block_number, position).await {
-			Ok(cell) => {
-				log::debug!("Fetched cell {} from IPFS", position.reference());
-				fetched.push(cell);
-			},
-			Err(error) => {
-				// Skip cells that returns error when fetching from IPFS
-				log::debug!(
-					"Error fetching cell {} from IPFS: {}",
-					position.reference(),
-					error
-				);
-				unfetched.push(position.clone());
-				continue;
-			},
-		}
-	}
+	let res = join_all(
+		positions
+			.iter()
+			.map(|position| fetch_cell_from_ipfs(ipfs, peers.clone(), block_number, position))
+			.collect::<Vec<_>>(),
+	)
+	.await;
+
+	let (fetched, unfetched): (Vec<_>, Vec<_>) = res
+		.into_iter()
+		.zip(positions)
+		.partition(|(res, _)| res.is_ok());
+
+	let fetched = fetched
+		.into_iter()
+		.map(|e| {
+			let cell = e.0.unwrap();
+			log::debug!("Fetched cell {} from IPFS", cell.reference());
+			cell
+		})
+		.collect::<Vec<_>>();
+
+	let unfetched = unfetched
+		.into_iter()
+		.map(|e| {
+			log::debug!(
```

Comment:
Maybe log it at error level?

Reply:
That could trigger alerts, not sure if we should discriminate errors from cells missing from IPFS.